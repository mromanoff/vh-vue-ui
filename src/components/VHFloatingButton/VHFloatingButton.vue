<script type="text/jsx">
export default {
  name: 'VhFloatingButton',
  functional: true,
  props: {
    /**
     * Button will become either "small" or "medium"
     */
    size: {
      type: String,
      default: 'medium',
      validator: value => {
        return value.match(/(small|medium)/);
      },
    },
  },

  render(h, { props, data, listeners, slots }) {
    const classes = {
      VHFloatingButton: true,
      [data.staticClass]: data.staticClass,
      [`VHFloatingButton--${props.size}`]: props.size,
    };

    return (
      <button
        class={classes}
        {...data}
        href={!props.nuxt && props.href ? props.href : null}
        to={props.nuxt && props.href}
        type={props.nuxt && props.href ? props.type : null}
      >
        {slots().default}
      </button>
    );
  },
};
</script>

<style scoped>
.VHFloatingButton {
  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--VHFloatingButton-color);
  background: var(--VHFloatingButton-backgroundColor);
  border-radius: var(--VHFloatingButton-borderRadius);
  user-select: none;
  text-decoration: none;
  outline: none;
  cursor: pointer;
  line-height: 0;
  -webkit-appearance: button;
  box-shadow: 0 3px 5px -1px rgba(0, 0, 0, 0.2), 0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12);
  transition: background-color 100ms ease-out 100ms;
  will-change: box-shadow;
  min-width: 0;
  padding: 0;
}

.VHFloatingButton:focus {
  outline: 0;
}

.VHFloatingButton:hover {
  background-color: var(--VHFloatingButton-backgroundColor-hover);
}

.VHFloatingButton:focus,
.VHFloatingButton:active {
  background-color: var(--VHFloatingButton-backgroundColor-active);
}

.VHFloatingButton--small {
  width: 24px;
  height: 24px;
}

.VHFloatingButton--medium {
  width: 48px;
  height: 48px;
}
</style>
