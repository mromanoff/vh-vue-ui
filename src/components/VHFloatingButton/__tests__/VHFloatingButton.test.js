import { shallowMount } from '@vue/test-utils';
import VHFloatingButton from '../VHFloatingButton.vue';

describe('VHFloatingButton.vue', () => {
  let wrapper;

  // TEST functional component with passing props: https://vue-test-utils.vuejs.org/api/options.html#context

  beforeEach(() => {
    wrapper = shallowMount(VHFloatingButton, {
      context: {
        props: { size: 'medium' },
      },
      slots: {
        default: '0',
      },
    });
  });

  it('renders default button', () => {
    expect(wrapper.isFunctionalComponent).toBe(true);
    expect(wrapper.is('button')).toBe(true);
    expect(wrapper.classes()).toStrictEqual(['VHFloatingButton', 'VHFloatingButton--medium']);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders small button', () => {
    wrapper = shallowMount(VHFloatingButton, {
      context: {
        props: { size: 'small' },
      },
      slots: {
        default: '0',
      },
    });
    expect(wrapper.is('button')).toBe(true);
    expect(wrapper.classes()).toStrictEqual(['VHFloatingButton', 'VHFloatingButton--small']);
    expect(wrapper).toMatchSnapshot();
  });
});
