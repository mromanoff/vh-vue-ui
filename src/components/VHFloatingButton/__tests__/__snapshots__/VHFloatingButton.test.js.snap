// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHFloatingButton.vue renders default button 1`] = `<button class="VHFloatingButton VHFloatingButton--medium">0</button>`;

exports[`VHFloatingButton.vue renders small button 1`] = `<button class="VHFloatingButton VHFloatingButton--small">0</button>`;
